### PR TITLE
Add mza banner to user edit page

### LINKED ISSUE #12469: Create banner for MzA change, followup to #12407
There should be a banner at the top that states this. Also when the time passes, the banner should turn past tense.

image

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDFRNT-239)

### PR DESCRIPTION
**Pull request by GregSherman**
Fixes #12469 
Adds a banner to the top of the page warning the user of the name switch. 
Screenshot 2023-10-18 at 12 06 27 PM

Learn more brings you to [here](https://support.mozilla.org/en-US/kb/change-primary-email-address-firefox-accounts).

After the date, the title renames to "Firefox Accounts was renamed ...".

This branch relies on pr #12466

### CODE REVIEW

**Comment by GregSherman** [comment]
@eviljeff requesting review

**Comment by eviljeff** [comment]
> Learn more brings you to [here](https://support.mozilla.org/en-US/kb/change-primary-email-address-firefox-accounts).

this is the url:
https://support.mozilla.org/kb/firefox-accounts-renamed-mozilla-accounts

**Comment by GregSherman** [comment]
> > Learn more brings you to [here](https://support.mozilla.org/en-US/kb/change-primary-email-address-firefox-accounts).
> 
> 
> 
> this is the url:
> 
> https://support.mozilla.org/kb/firefox-accounts-renamed-mozilla-accounts

Whoops, must've copied the wrong one

File: src/amo/pages/UserProfileEdit/index.js [hunk 1]
```diff
@@ -542,8 +543,66 @@ export class UserProfileEditBase extends React.Component<InternalProps, State> {
...
+        boldStart: '%(boldStart)s',
+        boldEnd: '%(boldEnd)s',
+      }),
+      replacements,
+    });
+
     return (
       <Page>
+        <div className="warning-container">
+          <Notice type="warning" key="MzA-branding" className="MzA-branding">
```
Comment (eviljeff) [inline src/amo/pages/UserProfileEdit/index.js:602]: nit: the className isn't consistently named with other classes on the page.  It'd be something like `UserProfileEdit-MzA-banner`

Comment (eviljeff) [reply src/amo/pages/UserProfileEdit/index.js:602]: another nit: do you need `key`?  It's not in a list.

**Review by eviljeff — CHANGES_REQUESTED** [review]
Just a minor nit.

**Review by willdurand — CHANGES_REQUESTED** [review]
I left some comments but here is a diff to apply in order to get most of the changes requested:

```diff
diff --git a/src/amo/pages/UserProfileEdit/index.js b/src/amo/pages/UserProfileEdit/index.js
index d1e2d55d1..6e44120dc 100644
--- a/src/amo/pages/UserProfileEdit/index.js
+++ b/src/amo/pages/UserProfileEdit/index.js
@@ -543,67 +543,32 @@ export class UserProfileEditBase extends React.Component<InternalProps, State> {
     const userProfileURL = `/user/${userId}/`;
     const overlayClassName = 'UserProfileEdit-deletion-modal';
 
-    const bannerText = isMzaBranding()
+    const title = isMzaBranding()
       ? i18n.gettext(
-          '%(boldStart)sFirefox Accounts was renamed to Mozilla accounts on Nov 1%(boldEnd)s %(newLine)s %(newLineEnd)sYou will still sign in with the same username and password, and there are no other changes to the products that you use. %(linkStart)sLearn More%(linkEnd)s',
+          'Firefox Accounts was renamed to Mozilla accounts on Nov 1',
         )
       : i18n.gettext(
-          '%(boldStart)sFirefox Accounts will be renamed to Mozilla accounts on Nov 1%(boldEnd)s %(newLine)s %(newLineEnd)sYou will still sign in with the same username and password, and there are no other changes to the products that you use. %(linkStart)sLearn More%(linkEnd)s',
+          'Firefox Accounts will be renamed to Mozilla accounts on Nov 1',
         );
 
-    const replacements = [
-      [
-        'linkStart',
-        'linkEnd',
-        (text) => {
-          return (
-            <a
-              target="_blank"
-              href="https://support.mozilla.org/kb/firefox-accounts-renamed-mozilla-accounts"
-              rel="noreferrer"
-              className="fxa-info-link"
-            >
-              {text}
-            </a>
-          );
-        },
-      ],
-      [
-        'newLine',
-        'newLineEnd',
-        () => {
-          return <br />;
-        },
-      ],
-      [
-        'boldStart',
-        'boldEnd',
-        (text) => {
-          return <strong>{text}</strong>;
-        },
-      ],
-    ];
-
-    const bannerContent = replaceStringsWithJSX({
-      text: i18n.sprintf(bannerText, {
-        linkEnd: '%(linkEnd)s',
-        linkStart: '%(linkStart)s',
-        newLine: '%(newLine)s',
-        newLineEnd: '%(newLineEnd)s',
-        boldStart: '%(boldStart)s',
-        boldEnd: '%(boldEnd)s',
-      }),
-      replacements,
-    });
-
     return (
       <Page>
-        <div className="warning-container">
-          <Notice type="warning" key="MzA-branding" className="MzA-branding">
-            <span className="notice-text-span">{bannerContent}</span>
-          </Notice>
-        </div>
-        {alternateOutput || (
+        {alternateOutput || [
+          <div className="UserProfileEdit-notice-wrapper">
+            <Notice
+              actionHref="https://support.mozilla.org/kb/firefox-accounts-renamed-mozilla-accounts"
+              actionTarget="_blank"
+              actionText={i18n.gettext('Learn more')}
+              className="UserProfileEdit-fxa-notice"
+              type="warning"
+            >
+              <strong>{title}</strong>
+              <br />
+              {i18n.gettext(`You will still sign in with the same username and
+                password, and there are no other changes to the products that
+                you use.`)}
+            </Notice>
+          </div>,
           <div className="UserProfileEdit">
             {user && (
               <Helmet>
@@ -966,8 +931,8 @@ export class UserProfileEditBase extends React.Component<InternalProps, State> {
                 </div>
               </OverlayCard>
             )}
-          </div>
-        )}
+          </div>,
+        ]}
       </Page>
     );
   }
diff --git a/src/amo/pages/UserProfileEdit/styles.scss b/src/amo/pages/UserProfileEdit/styles.scss
index 8c40a5ad8..ed9a5ebdb 100644
--- a/src/amo/pages/UserProfileEdit/styles.scss
+++ b/src/amo/pages/UserProfileEdit/styles.scss
@@ -2,20 +2,16 @@
 
 $font-size-aside: $font-size-xs;
 
-.UserProfileEdit {
+.UserProfileEdit-notice-wrapper {
   @include page-padding;
-
-  @include respond-to(large) {
-    display: flex;
-    justify-content: flex-start;
-  }
 }
 
-.warning-container {
+.UserProfileEdit {
   @include page-padding;
 
   @include respond-to(large) {
-    margin-bottom: -$padding-page;
+    display: flex;
+    justify-content: flex-start;
   }
 }
 
```

**Comment by willdurand** [comment]
This is how it is going to look like:

Screenshot 2023-10-18 at 23 20 06

**Comment by willdurand** [comment]
I also filed https://github.com/mozilla/addons-frontend/issues/12479, which would be good to address as a follow-up (since this patch was already passing `rel="noopener"`).

File: src/amo/pages/UserProfileEdit/index.js (hunk 1 above)
Comment (GregSherman) [reply src/amo/pages/UserProfileEdit/index.js]: well, now it is!

**Review by willdurand — APPROVED** [review]
This is looking good to me, thanks!

It'd be good if you could address https://github.com/mozilla/addons-frontend/issues/12479 later since that'd be kinda nice to have.

**Pull request merged**